### PR TITLE
feat(ModelArts): add new resource to manage notebook mount

### LINKED ISSUE #1875: [modelArts] Add a resource to Mount OBS to notebook
Add a resource to Mount OBS to notebook

### Current Terraform Version

```
...
```

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
add new resource to manage notebook mount

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1875 

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/modelarts' TESTARGS='-run=TestAccResourceNotebookMountStorage_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/modelarts -v -run=TestAccResourceNotebookMountStorage_basic -timeout 360m -parallel 4
=== RUN   TestAccResourceNotebookMountStorage_basic
=== PAUSE TestAccResourceNotebookMountStorage_basic
=== CONT  TestAccResourceNotebookMountStorage_basic
--- PASS: TestAccResourceNotebookMountStorage_basic (123.67s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/modelarts 123.767s
```

### CODE REVIEW

File: docs/resources/modelarts_notebook_mount_storage.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) The region in which to create the resource. If omitted, the
+  provider-level region will be used. Changing this parameter will create a new resource.
+
+* `notebook_id` - (Required, String, ForceNew) Specifies ID of notebook which the storage be mounted to.
+ Changing this parameter will create a new resource.
+
+* `uri` - (Required, String, ForceNew) Specifies the uri of Parallel File System (PFS) or its folders in OBS.
```
Comment (ShiChangkuo) [inline docs/resources/modelarts_notebook_mount_storage.md]: uri or storage_path ?

File: docs/resources/modelarts_notebook_mount_storage.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  Default value is `OBSFS`,
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID in format of **notebook_id/mount_id**. It is composed of the ID of notebook and mount ID,
+ separated by a slash.
+
+* `mount_id` - The mount ID.
```
Comment (ShiChangkuo) [inline docs/resources/modelarts_notebook_mount_storage.md:45]: suggest to use mount_id as the resource id.

File: huaweicloud/services/modelarts/resource_huaweicloud_modelarts_notebook_mount_storage.go [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile("^/data/[a-zA-Z0-9_-]*/$"),
+					"Only the sub directory of `/data/`can be mounted. e.g. /data/dir1/"),
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "OBSFS",
```
Comment (ShiChangkuo) [inline huaweicloud/services/modelarts/resource_huaweicloud_modelarts_notebook_mount_storage.go]: suggest to add a validation or just set it as an attribute.

File: huaweicloud/services/modelarts/resource_huaweicloud_modelarts_notebook_mount_storage.go [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+
+	return nil
+}
+
+func waitingNotebookForMount(ctx context.Context, client *golangsdk.ServiceClient, notebookId, mountId string,
+	timeout time.Duration) error {
+	createStateConf := &resource.StateChangeConf{
+		Pending: []string{"MOUNTING"},
+		Target:  []string{"MOUNTED"},
+		Refresh: func() (interface{}, string, error) {
```
Comment (ShiChangkuo) [inline huaweicloud/services/modelarts/resource_huaweicloud_modelarts_notebook_mount_storage.go:184]: I think you can merge the refresh function between waitingNotebookForMount and waitingNotebookMountForDeleted

**Comment by ShiChangkuo** [comment]
please attach the accptance tests

File: huaweicloud/services/modelarts/resource_huaweicloud_modelarts_notebook_mount_storage.go (hunk 3 above)
Comment (hmbig2) [reply huaweicloud/services/modelarts/resource_huaweicloud_modelarts_notebook_mount_storage.go]: OK ， set as a export attribute

File: huaweicloud/services/modelarts/resource_huaweicloud_modelarts_notebook_mount_storage.go (hunk 4 above)
Comment (hmbig2) [reply huaweicloud/services/modelarts/resource_huaweicloud_modelarts_notebook_mount_storage.go:184]: pend, target and refresh  is different，too few in common

File: docs/resources/modelarts_notebook_mount_storage.md (hunk 2 above)
Comment (hmbig2) [reply docs/resources/modelarts_notebook_mount_storage.md:45]: This format is more convenient for users.

File: docs/resources/modelarts_notebook_mount_storage.md (hunk 1 above)
Comment (hmbig2) [reply docs/resources/modelarts_notebook_mount_storage.md]: change to storage_path

**Comment by niuzhenguo** [comment]
Pls move this to docs beta this release.
https://github.com/huaweicloud/terraform-provider-huaweicloud/pull/1986

**Pull request merged**